### PR TITLE
Upgrade guide for subscription storage

### PR DESCRIPTION
**Pull request by danielmarbach**
Belongs to https://github.com/Particular/NServiceBus/pull/2804

### CODE REVIEW

**Comment by danielmarbach** [comment]
@DavidBoike the doco is here

File: nservicebus/upgrades/5to6.md [hunk 1]
```diff
@@ -105,6 +105,13 @@ NServiceBus will automatically make correlated saga properties unique without th
 Saga persisters (`ISagaPersister`) and finders (`IFindSagas`) introduce a new parameter `SagaPersistenceOptions`. This parameter give access to the saga metadata and pipeline context. This enables  persisters and finders to manipulate everything that exists in the context during message pipeline execution. For more information see [Sagas](/nservicebus/sagas/) and [Complex saga finding logic](/nservicebus/sagas/saga-finding.md).
 
 
+## Subscriptions
+
+### Subscription storage
+`ISubscriptionStorage` has been split into three interfaces, `ISubscriptionStorage`, `IInitializableSubscriptionStorage` and `IQuerySubscriptions`, to properly separate those storage concerns. `ISubscriptionStorage` and `IQuerySubscriptions` must be implemented to have a fully functional subscription infrastructure. `IInitializableSubscriptionStorage` is only necessary when the subscription storage needs to be initialized.
```
Comment (SimonCropp) [inline nservicebus/upgrades/5to6.md]: So while this is good from the upgrade perspective where is the full doco for someone who starts on version 6. Where do they read about what these interfaces are for?

**Comment by DavidBoike** [comment]
@SimonCropp is right - we need to update [Authoring a custom persistence](http://docs.particular.net/nservicebus/persistence/authoring-custom) too.

Simon, right now the interfaces (that are breaking) are rendered by Markdown. Would you recommend moving that to snippets, and then in the V6 snippet showing more than one interface in that snippet? Or some other way?

**Comment by DavidBoike** [comment]
This PR is merged, move discussion to #757

**Pull request merged**